EIPS/eip-xxxx.md

```diff
@@ -0,0 +1,136 @@
+
+## Motivation
+
+Currently, every Ethereum block includes two `state_root`. 
+
+The consensus layer (CL) `state_root` contained within the `BeaconBlock` tracks CL state (e.g., validator entries and exits).
+
+The execution layer (EL) `state_root` contained within the `ExecutionPayload` tracks the effect of transaction execution (e.g., account balances, code, etc.). 
+
+The EL `state_root` must be computed by block builders (and verified by relays, validators, and light clients) in near-real-time. This computation accounts for a large fraction of block production and verification time, especially in an MEV-Boost environment. This overhead is also a challenge for efforts to enable real-time ZK proving of the chain.
+
+By delaying the EL `state_root` reference by one block, we can remove EL `state_root` calculation overhead from the critical path of block production. Instead, clients can pipeline calculation of block `n-1`'s EL `state_root` during the idle slot time of block `n`.
+
+This change will lower latency at the tip, allowing us to increase throughput and simplify the block production pipeline. It can also significantly accelerate the timeline real-time ZK proving Ethereum.
+
+## Specification
+
+### Execution Layer Changes
+
+1. **`ExecutionPayload` State Root Field**  
+   - **Before** (in the current post-Merge design):  
+     ```solidity
+     struct ExecutionPayload {
+       uint256 blockNumber;
+       uint256 timestamp;
+       bytes32 parentHash;
+       address feeRecipient;
+       ...
+       bytes32 stateRoot;  // references the post-state root of *this* block's transactions
+       ...
+     }
+     ```
+   - **After** (conceptual):  
+     ```solidity
+     struct ExecutionPayload {
+       uint256 blockNumber;
+       uint256 timestamp;
+       bytes32 parentHash;
+       address feeRecipient;
+       ...
+       bytes32 stateRoot;  // now references the post-state root of block (n-1)
+       ...
+     }
+     ```
+    - In other words, block `n`’s `ExecutionPayload.stateRoot` must now point to the **execution-layer post-state of block `n-1`**, rather than referencing the new state after block `n`’s own transactions.
+
+2. **Block Header Validation**  
+   - **Remove** the rule requiring each block’s `stateRoot` to match the **post-state** produced by the transactions in *that same block*:  
+   - **Instead**, require that `payload_n.stateRoot == post_state(n-1)`:
+     ```python
+     def validate_execution_payload(payload_n, post_state_of_block_n_minus_1):
+         # Was:
+         #   assert compute_state_root(transactions_of_block_n) == payload_n.stateRoot
+         #
+         # Now:
+         assert payload_n.stateRoot == post_state_of_block_n_minus_1
```

Comment:
Does this mean that if a future protocol upgrade changes EVM behavior, then nodes that did not upgrade would fork off the network with 1 block delay? (Though I don't think this is an issue.)

Reply:
I guess that is true only if the protocol change results in a mismatched state but doesn't change the execution flow of the transactions.

Reply:
Right, so for instance adding a new tx type could result in a fork exactly at the upgrade block.

But adding a new opcode or precompile, or changing the gas schedule, could result in a fork one block later.